clang/lib/CodeGen/CGHLSLRuntime.h

```diff
@@ -87,6 +87,7 @@ class CGHLSLRuntime {
   GENERATE_HLSL_INTRINSIC_FUNCTION(SDot, sdot)
   GENERATE_HLSL_INTRINSIC_FUNCTION(UDot, udot)
   GENERATE_HLSL_INTRINSIC_FUNCTION(WaveIsFirstLane, wave_is_first_lane)
+  GENERATE_HLSL_INTRINSIC_FUNCTION(WaveReadLaneAt, waveReadLaneAt)
```

Comment:
nit: Intrinsic declarations are all `_` separated, not `camelCase`.

Reply:
If we use `_` separated then the generated names will use `.` to separate them. So eg `wave_read_lane_at` becomes `llvm.dx.wave.read.lane.at`. Which we decided was quite confusing due to the confusion with a namespace. I will clean up the other intrinsic to cohere to this in a follow up pr.

Reply:
This isn't just a convention for our backend, this is a convention common across LLVM backends. For example look at the AMD GCN intrinsics like `llvm.amdgcn.raw.ptr.buffer.store`.

I can see an argument that the `_`s were placed in the wrong places, but I think this was the wrong decision. A better name might be `wave_readlaneat`, but `camelCase` isn't really used in intrinsic names.

@bogner & @farzonl, we should discuss this before we break with common LLVM convention.

Reply:
The equivalent AMD GPU intrinsic is named `int_amdgcn_readlane`, which translates to `llvm.amdgcn.readlane`. Having `dx.readlane` or `dx.wave.readlane` seems like a preferable naming approach.

Reply:
This is my bad. Finn was using the llvm convention. I didn’t like the way it looked.

Reply:
Updated to `wave_readlaneat`.